File: PASCAL/Micro10.pas
program micro10;

var 
numero: integer;
fat : integer;

function Fatorial (numero :integer) : integer;
begin
    if (numero <= 0) then
        Fatorial := 1
    else
        Fatorial := numero * Fatorial(numero -1);
end;

begin
    write('Digite um numero: ');
    readln(numero);
    fat := Fatorial(numero);
    write('O fatorial de ');
    write(numero);
    write(' e ');
    writeln(fat);

end.

File: IR LLVM/Micro10/Micro10.c
#include <stdio.h>
#include <stdlib.h>

int fatorial(int n){
    if(n <= 0)
        return 1;
    return n * fatorial(n-1);
}

int main(){
    int num;
    int fat;
    printf("Digite um numero:");
    scanf("%d", &num);
    fat = fatorial(num);
    printf("O fatorial de %d e %d\n", num, fat);
    return 0;
}